subprojects/docs/src/samples/antlr/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,26 @@
+// tag::use-plugin[]
+plugins {
+    antlr
+}
+// end::use-plugin[]
+
+// tag::declare-dependency[]
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    antlr("org.antlr:antlr:3.5.2")   // use ANTLR version 3
+    // antlr("org.antlr:antlr4:4.5") // use ANTLR version 4
+// end::declare-dependency[]
+    testImplementation("junit:junit:4.12")
+// tag::declare-dependency[]
+}
+// end::declare-dependency[]
+
+// tag::generate-grammar-settings[]
+tasks.getByName<AntlrTask>("generateGrammarSource") {
+    maxHeapSize = "64m"
+    arguments = arguments + listOf("-visitor", "-long-messages")
```

Comment:
Note to reviewer:
Using the `+=` operator as in Groovy causes the following compilation error:
```
                         ^ Assignment operators ambiguity: 
                             public operator fun <T> Collection<String!>.plus(elements: Iterable<String!>): List<String!> defined in kotlin.collections
                             @InlineOnly public inline operator fun <T> MutableCollection<in String>.plusAssign(elements: Iterable<String>): Unit defined in kotlin.collections
```

Reply:
This is expected. The `arguments` property is mapped to a `kotlin.collections.List` which is not mutable so has no `+=` operator.

Reply:
Hmm, it seems it's not the case, otherwise there would be no ambguity: the message says that `MutableCollection<in String>.plusAssign()` is a valid candidate.
Am I missing something?

Reply:
Ah right, my mistake. It is mapped to `MutableList` indeed.

I get the same:
![image](https://user-images.githubusercontent.com/132773/44974979-87cf5300-af61-11e8-9259-83b061a5c28e.png)

which is explained at https://discuss.kotlinlang.org/t/compiler-error-assignment-operators-ambiguity-all-these-functions-match/5069